cloud: add changefeed docs for TiDB Cloud Essential (#22244)
pingcap/docs: tidb-cloud/essential-changefeed-sink-to-kafka.md

## Prerequisites

Before creating a changefeed to stream data to Apache Kafka, you need to complete the following prerequisites:

- Set up your network connection
- Add permissions for Kafka ACL authorization

### Network

Ensure that your {{{ .essential }}} cluster can connect to the Apache Kafka service. You can choose one of the following connection methods:

- Private Link Connection: meeting security compliance and ensuring network quality.
- Public Network: suitable for a quick setup.

<SimpleTab>
<div label="Private Link Connection">

Private link connections leverage **Private Link** technologies from cloud providers to enable resources in your VPC to connect to services in other VPCs using private IP addresses, as if those services were hosted directly within your VPC.

{{{ .essential }}} currently supports Private Link Connection only for self-hosted Kafka and Confluent Cloud Dedicated Clusters. It does not support direct integration with MSK, or other Kafka SaaS services.

> See the following instructions to set up a Private Link connection according to your Kafka deployment and cloud provider:

Reviewer: ```suggestion
To set up a Private Link connection based on your Kafka deployment and cloud provider, see the following guides:
```

## Step 1. Open the Changefeed page for Apache Kafka

1. Log in to the [TiDB Cloud console](https://tidbcloud.com).
2. Navigate to the overview page of the target TiDB Cloud cluster, and then click **Data** > **Changefeed** in the left navigation pane.
3. Click **Create Changefeed**, and then select **Kafka** as **Destination**.

## Step 2. Configure the changefeed target

The steps vary depending on the connectivity method you select.

<SimpleTab>
<div label="Public">

1. In **Connectivity Method**, select **Public**, and fill in your Kafka broker endpoints. You can use commas `,` to separate multiple endpoints.
2. Select an **Authentication** option according to your Kafka authentication configuration.

    - If your Kafka does not require authentication, keep the default option **Disable**.
    - If your Kafka requires authentication, select the corresponding authentication type, and then fill in the **user name** and **password** of your Kafka account for authentication.

> 3. Select your **Kafka Version**. Choose **Kafka v2** or **Kafka v3** according to the version of your Kafka.

Reviewer: ```suggestion
3. For **Kafka Version**, select **Kafka v2** or **Kafka v3** based on your Kafka version.
```

4. Select a **Compression** type for the data in this changefeed.
5. Enable the **TLS Encryption** option if your Kafka has enabled TLS encryption and you want to use TLS encryption for the Kafka connection.
6. Click **Next** to test the network connection. If the test succeeds, you will be directed to the next page.

</div>
<div label="Private Link">

1. In **Connectivity Method**, select **Private Link**.
2. In **Private Link Connection**, select the private link connection that you created in the [Network](#network) section. Make sure the Availability Zones of the private link connection match those of the Kafka deployment.
3. Fill in the **Bootstrap Port** that you obtained from the [Network](#network) section.
4. Select an **Authentication** option according to your Kafka authentication configuration.

    - If your Kafka does not require authentication, keep the default option **Disable**.
    - If your Kafka requires authentication, select the corresponding authentication type, and then fill in the **user name** and **password** of your Kafka account for authentication.

> 5. Select your **Kafka Version**. Choose **Kafka v2** or **Kafka v3** according to the version of your Kafka.

Reviewer: ```suggestion
5. For **Kafka Version**, select **Kafka v2** or **Kafka v3** based on your Kafka version.
```

6. Select a **Compression** type for the data in this changefeed.
7. Enable the **TLS Encryption** option if your Kafka has enabled TLS encryption and you want to use TLS encryption for the Kafka connection.

> 8. Input the **TLS Server Name** if your Kafka requires TLS SNI verification. For example, `Confluent Cloud Dedicated clusters`.

Reviewer: ```suggestion
8. If your Kafka requires TLS SNI verification, enter the **TLS Server Name**. For example, `Confluent Cloud Dedicated clusters`.
```

    - Debezium is a tool for capturing database changes. It converts each captured database change into a message called an "event" and sends these events to Kafka. For more information, see [Debezium data format](https://docs.pingcap.com/tidb/stable/ticdc-debezium).

5. Enable the **TiDB Extension** option if you want to add TiDB-extension fields to the Kafka message body.

    For more information about TiDB-extension fields, see [TiDB extension fields in Avro data format](https://docs.pingcap.com/tidb/stable/ticdc-avro-protocol#tidb-extension-fields) and [TiDB extension fields in Canal-JSON data format](https://docs.pingcap.com/tidb/stable/ticdc-canal-json#tidb-extension-field).

6. If you select **Avro** as your data format, you will see some Avro-specific configurations on the page. You can fill in these configurations as follows:

    - In the **Decimal** and **Unsigned BigInt** configurations, specify how TiDB Cloud handles the decimal and unsigned bigint data types in Kafka messages.

>     - In the **Schema Registry** area, fill in your schema registry endpoint. If you enable **HTTP Authentication**, the fields for user name and password are displayed to fill in.

Reviewer: ```suggestion
    - In the **Schema Registry** area, fill in your schema registry endpoint. If you enable **HTTP Authentication**, enter the user name and password.
```

9. In the **Topic Configuration** area, configure the following numbers. The changefeed will automatically create the Kafka topics according to the numbers.

    - **Replication Factor**: controls how many Kafka servers each Kafka message is replicated to. The valid value ranges from [`min.insync.replicas`](https://kafka.apache.org/33/documentation.html#brokerconfigs_min.insync.replicas) to the number of Kafka brokers.
    - **Partition Number**: controls how many partitions exist in a topic. The valid value range is `[1, 10 * the number of Kafka brokers]`.

10. In the **Split Event** area, choose whether to split `UPDATE` events into separate `DELETE` and `INSERT` events or keep as raw `UPDATE` events. For more information, see [Split primary or unique key UPDATE events for non-MySQL sinks](https://docs.pingcap.com/tidb/stable/ticdc-split-update-behavior/#split-primary-or-unique-key-update-events-for-non-mysql-sinks).

11. Click **Next**.

## Step 4. Review and create your changefeed

1. In the **Changefeed Name** area, specify a name for the changefeed.

> 2. Review all the changefeed configurations that you set. Click **Previous** to go back to the previous configuration pages if you want to modify some configurations. 

Reviewer: ```suggestion
2. Review all the changefeed configurations that you set. Click **Previous** to make changes if necessary.
```
